File: Basic_Algorithm/799/subsequence.cpp
#include <iostream>
using namespace std;

const int N = 1e5 + 10;
int n, res, a[N], cnt[N];

int main() {
    scanf("%d", &n);
    for (int i = 0; i < n; i++) scanf("%d", &a[i]);
    for (int i = 0, j = 0; i < n; i++) {
        cnt[a[i]]++;
        while (cnt[a[i]] > 1) cnt[a[j++]]--;
        res = max(res, i - j + 1);
    }
    printf("%d", res);
    return 0;
}

File: Basic_Algorithm/799/subsequence.py
if __name__ == "__main__":
    n = int(input())
    a = list(map(int, input().split()))
    cnt = [0] * (10 ** 5 + 10)
    i, j = 0, 0
    res = 0
    while j < n:
        cnt[a[j]] += 1
        while cnt[a[j]] > 1:
            cnt[a[i]] -= 1
            i += 1
        res = max(res, j - i + 1)
        j += 1
    print(res)
